consensus/src/user/provisioners.rs

```diff
@@ -12,17 +12,55 @@ use node_data::bls::{PublicKey, PublicKeyBytes};
 use node_data::ledger::Seed;
 use num_bigint::BigInt;
 use std::collections::BTreeMap;
+use std::mem;
 
 use super::committee::Committee;
 
 pub const DUSK: u64 = 1_000_000_000;
 
-#[derive(Clone, Default, Debug)]
+#[derive(Clone, Debug)]
 pub struct Provisioners {
     members: BTreeMap<PublicKey, Stake>,
 }
 
+#[derive(Clone, Debug)]
+pub struct ContextProvisioners {
+    current: Provisioners,
+    prev: Option<Provisioners>,
+}
+
+impl ContextProvisioners {
+    pub fn new(current: Provisioners) -> Self {
+        Self {
+            current,
+            prev: None,
+        }
+    }
+    pub fn current(&self) -> &Provisioners {
+        &self.current
+    }
+    pub fn to_current(&self) -> Provisioners {
+        self.current.clone()
+    }
+    pub fn prev(&self) -> &Provisioners {
+        self.prev.as_ref().unwrap_or(&self.current)
+    }
+    pub fn replace(&mut self, mut current: Provisioners) {
```

Comment:
Readability would probably benefit by choosing a different parameter name :) (e.g. `new`)

Reply:
I called it `current` because it changes what `.current()` returns

Reply:
yes, but you're effectively updating the current list and setting prev to current.
Anyway, I'll let you decide.